File: data/test00.hs
{-|
Module      : Main
Description : Module-level documentation for Coding Agent testing.
-}
module Main where

import Data.Map (Map)
import qualified Data.Map as Map

-- | Manages user session and authentication (Haskell records mimicking class storage)
data UserManager = UserManager 
  { dbUrl :: String 
  }

getUserById :: UserManager -> Int -> IO (Maybe (Map String String))
getUserById manager userId = do
  -- TODO: Implement database lookup
  return Nothing

main :: IO ()
main = do
  -- パターンマッチングのテスト
  let status = 200 :: Int
  case status of
    200 -> putStrLn "Success"
    201 -> putStrLn "Success"
    _   -> putStrLn "Failure"

File: data/test00.rb
# frozen_string_literal: true

# Module-level documentation for Coding Agent testing.

# Manages user session and authentication.
class UserManager
  attr_reader :db_url

  def initialize(db_url)
    @db_url = db_url
  end

  def get_user_by_id(user_id)
    # TODO: Implement database lookup
    nil
  end
end

def main
  # Ruby 3.0+ パターンマッチング (case in) のテスト
  status = 200
  case status
  in 200 | 201
    puts "Success"
  else
    puts "Failure"
  end
end

if __FILE__ == $PROGRAM_NAME
  main
end
